README.md

```diff
@@ -10,3 +10,14 @@ abandoned completely and merged into node_exporter**
 This currently exposes all basic stats (vdev_stats) and most extended stats (vdev_stats_ex) on a
 vdev-level. It doesn't expose per-disk or per-zpool stats, even though these are also available from
 the underlying API.
+
+
+To build with version info:
+
+    docker run -it --rm -v "$PWD":/usr/src/zfs_exporter -w /usr/src/zfs_exporter golang:1.19-bullseye go build -v \
+        -ldflags="-s -w
+            -X github.com/prometheus/common/version.Version=$(git describe)
+            -X github.com/prometheus/common/version.BuildUser=$(whoami)
+            -X github.com/prometheus/common/version.BuildDate=$(date +%Y-%m-%d)
+            -X github.com/prometheus/common/version.Revision=$(git rev-parse --short HEAD)
+            -X github.com/prometheus/common/version.Branch=$(git branch | sed -n -e 's/^\* \(.*\)/\1/p')"
```

Comment:
Where does the magic regex come from?

Reply:
[stackoverflow](https://stackoverflow.com/a/2111099)
`-n` is to not print anything by default and the `p` modifier is to print replaced contents. This will select the line starting with `*` (the active branch) and print its end. Parenthesis are back-quoted in standard sed regexp to mean they are capturing, not literal characters.

git branch output:
````
* master
another-branch
````

Reply:
Not always: when checkouting a tag it doesn't work:

````
$ git tag dummy f85e19b5ef8d6f49330ae66fda1a726d2679a77a
$ git co dummy 
HEAD est maintenant sur f85e19b Init
$ git symbolic-ref --short HEAD
fatal: ref HEAD is not a symbolic ref
$ git branch | sed -n -e 's/^\* \(.*\)/\1/p'
(HEAD detached at dummy)
````